src/primitives/utils/withScrolling.ts

```diff
@@ -156,6 +163,22 @@ export function withScrolling(isRow: boolean) {
 
     // Update position if it has changed
     if (componentRef[axis] !== nextPosition) {
+      if (
+        onScrolled &&
+        componentRef[axis] === componentRef._initialPosition &&
+        componentRef._initialPosition !== nextPosition
+      ) {
```

Comment:
a !== b
a === c && c !== b

if a = c then a !== b is rechecking the initial if statement